src/Stache/Query/Builder.php

```diff
@@ -131,14 +131,55 @@ protected function filterWhereBasic($values, $where)
     protected function filterWhereIn($values, $where)
     {
         return $values->filter(function ($value) use ($where) {
+            if (is_array($value)) {
+                return count(array_intersect($value, $where['values']));
+            }
+
             return in_array($value, $where['values']);
         });
     }
 
     protected function filterWhereNotIn($values, $where)
     {
         return $values->filter(function ($value) use ($where) {
+            if (is_array($value)) {
+                return ! count(array_intersect($value, $where['values']));
+            }
+
             return ! in_array($value, $where['values']);
         });
     }
+
+    protected function filterWhereJsonContains($values, $where)
+    {
+        return $values->filter(function ($value) use ($where) {
+            if (! is_array($value)) {
+                return false;
+            }
+
+            return count(array_intersect($value, $where['values']));
```

Comment:
`return ! empty(array_intersect($value, $where['values']));`

Reply:
Functionally does the same thing - is there a reason for the preference to empty()? I can see other examples of `! count()` in the codebase

Reply:
The main reason is to return the correct type, boolean, the second is that `empty()` is inherently faster than `count()`. With `! count()` you're also doing a magic conversion from `int` to `bool`, so while it's negligible it's still additional logic PHP has to do

Reply:
Benchmarks I've read says count() on an array is pretty much the same as empty() in terms of performance. I dont mind changing it, I just dont think its as clear cut as you do :)